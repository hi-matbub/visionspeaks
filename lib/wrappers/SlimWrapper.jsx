import React from 'react';
import PropTypes from 'prop-types';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import clsx from 'clsx';

const SlimWrapper = ({ children, className }) => (
  <Row className={clsx('align-items-center justify-content-center', className)}>
    <Col lg={8} md={9} sm={12}>
      {children}
    </Col>
  </Row>
);

SlimWrapper.propTypes = {
  children: PropTypes.node.isRequired,
  // eslint-disable-next-line react/require-default-props
  className: PropTypes.string,
};

export default SlimWrapper;
