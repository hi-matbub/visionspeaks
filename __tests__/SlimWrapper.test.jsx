import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SlimWrapper from '../lib/wrappers/SlimWrapper';

it('SlimWrapper component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SlimWrapper>Hello, World!</SlimWrapper>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('SlimWrapper component renders unchanged', () => {
  const tree = renderer.create(<SlimWrapper>Hello, World!</SlimWrapper>).toJSON();
  expect(tree).toMatchSnapshot();
});
