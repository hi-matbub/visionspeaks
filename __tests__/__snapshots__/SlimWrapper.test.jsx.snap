// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlimWrapper component renders unchanged 1`] = `
<div
  className="align-items-center justify-content-center row"
>
  <div
    className="col-lg-8 col-md-9 col-sm-12"
  >
    Hello, World!
  </div>
</div>
`;
